src/python/TaskWorker/Actions/DBSDataDiscovery.py

```diff
@@ -411,8 +412,75 @@ def executeInternal(self, *args, **kwargs):
 
         self.logger.debug("Got %s files", len(result.result.getFiles()))
 
+        # lock input data on disk before final submission (only for CMS datasets !)
+        # (submission can still fail in Splitting step, but it happens rarely and usually users fix
+        # a configuration mistake submit, i.e. they really want these data, so let's lock anyhow
+
+        if not isUserDataset:
+            try:
+                if not inputBlocks:
+                    self.lockInputData(dataToLock=inputDataset)
+                else:
+                    self.lockInputData(dataToLock=inputBlocks)
+            except Exception:
+                msg = 'Locking of input data failed. Details in TaskWorker log. Submit anyhow'
+                self.logger.info(msg)
+                self.uploadWarning(msg, self.userproxy, self.taskName)
+
         return result
 
+
+    def lockInputData(self, dataToLock=None):
+        """
+        lock input data on disk via a Rucio rule
+        """
+        #TODO should use a different account than crab_tape_recall. WIll ask Rahul once code is tested
+        #TODO a lot of code is in common with requestTapeRecall and requestRucioTapeRecall must fix this
+        if not isinstance(dataToLock, str) and not isinstance(dataToLock, list):
+            return
+
+        msg = f"Lock {dataToLock} on disk"
+        self.logger.info(msg)
+        # need to a "privileged" Rucio account to create containers and create rules
+        rucioAccount = 'crab_input'
+        lockConfig = copy.deepcopy(self.config)
+        lockConfig.Services.Rucio_account = rucioAccount
+        rucioClient = getNativeRucioClient(lockConfig, self.logger)  # pylint: disable=redefined-outer-name
+
+        # prepare container to be locked
+        if isinstance(dataToLock, str):
+            # recalling a full DBS dataset, simple the container already exists
+            myScope = 'cms'
+            dbsDatasetName = dataToLock
+            containerDid = {'scope': myScope, 'name': dbsDatasetName}
+        elif isinstance(dataToLock, list):
+            # need to prepare ad hoc container. Can not reuse a possibly existing one
+            # for this dataset which may have had a different content
+            # Therefore we stick with naming the container after the task name
+            myScope = 'user.crab_tape_recall'  # do not mess with cms scope
+            containerName = '/TapeRecall/%s/USER' % self.taskName.replace(':', '.')
```

Comment:
Is this your intention to use the scope/name same as tape recall?

Reply:
well spotted. thanks. I will use new account `crab_input` here as well.

Reply:
fixed in https://github.com/dmwm/CRABServer/pull/7732/commits/e04e165ea3e1a31d8c2190431b596d278b408815